app/components/CleanComponent.js

```diff
@@ -0,0 +1,135 @@
+// @flow
+import React, { Component } from "react";
+import {
+  Grid,
+  Button,
+  Icon,
+  Segment,
+  Header,
+  Dropdown
+} from "semantic-ui-react";
+import { isNil } from "lodash";
+import styles from "./styles/common.css";
+import {
+  EXPERIMENTS,
+  DEVICES,
+  KERNEL_STATUS
+} from "../constants/constants";
+import { Kernel } from "../constants/interfaces";
+import { readEEGDataDir } from "../utils/filesystem/write";
+
+interface Props {
+  type: ?EXPERIMENTS;
+  deviceType: DEVICES;
+  mainChannel: ?any;
+  kernel: ?Kernel;
+  epochsInfo: ?{ [string]: number };
+  jupyterActions: Object;
+}
+
+interface State {
+  eegFilePaths: Array<?{
+    key: string,
+    text: string,
+    value: { name: string, dir: string }
+  }>;
+  selectedFilePaths: Array<?string>;
+}
+
+export default class Clean extends Component<Props, State> {
+  props: Props;
+  state: State;
+  handleDropdownChange: (Object, Object) => void;
+  handleLoadData: () => void;
+
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      eegFilePaths: [{ key: "", text: "", value: { name: "", dir: "" } }],
+      selectedFilePaths: []
+    };
+    this.handleDropdownChange = this.handleDropdownChange.bind(this);
+    this.handleLoadData = this.handleLoadData.bind(this);
+  }
+
+  componentWillMount() {
+    if (isNil(this.props.kernel)) {
```

Comment:
Why not load the kernel when the app is launched?

Reply:
It seems odd to launch a secondary process right away for something that won't get used until maybe 10 minutes in. There's also the chance that many uses of the app will be focused on viewing EEG, practicing experiments, or just collecting data and won't need Jupyter at all. There may be a performance benefit as well to launching it as needed.